thirdparty/github.go

```diff
@@ -1260,39 +1044,21 @@ func AppAuthorizedForOrg(ctx context.Context, token, requiredOrganization, name
 }
 
 func GitHubUserPermissionLevel(ctx context.Context, token, owner, repo, username string) (string, error) {
-	installationToken, err := GetInstallationToken(ctx, owner, repo, nil)
-	if err != nil {
-		ctx, httpClient, putClient := getGithubClient(ctx, installationToken, "GithubUserPermissionLevel", retryConfig{retry404: true}, []attribute.KeyValue{
-			attribute.String(githubOwnerAttribute, owner),
-			attribute.String(githubRepoAttribute, repo),
-		})
-		defer putClient()
-		client := github.NewClient(httpClient)
-
-		permissionLevel, _, err := client.Repositories.GetPermissionLevel(ctx, owner, repo, username)
-		if err == nil && permissionLevel != nil && permissionLevel.Permission != nil {
-			return permissionLevel.GetPermission(), nil
-		}
-
-		grip.Debug(message.WrapError(err, message.Fields{
-			"ticket":  "EVG-19966",
-			"message": "failed to get permissions from GitHub",
-			"caller":  "GithubUserPermissionLevel",
-			"owner":   owner,
-			"repo":    repo,
-			"user":    username,
-		}))
-	}
-
-	// Fallback to not using the GitHub app on error.
-	ctx, legacyClient, putLegacyClient := getGithubClient(ctx, token, "GithubUserPermissionLevel", retryConfig{retry404: true}, []attribute.KeyValue{
+	caller := "GithubUserPermissionLevel"
+	ctx, span := tracer.Start(ctx, caller, trace.WithAttributes(
+		attribute.String(githubEndpointAttribute, caller),
 		attribute.String(githubOwnerAttribute, owner),
 		attribute.String(githubRepoAttribute, repo),
-	})
-	defer putLegacyClient()
-	client := github.NewClient(legacyClient)
+	))
+	defer span.End()
+
+	githubClient := getGithubClient(getInstallationToken(ctx, token, owner, repo, nil), caller, retryConfig{retry404: true})
 
-	permissionLevel, _, err := client.Repositories.GetPermissionLevel(ctx, owner, repo, username)
+	permissionLevel, resp, err := githubClient.Repositories.GetPermissionLevel(ctx, owner, repo, username)
```

Comment:
Do we maybe not want to use the cached value here since using an outdated value here can have bigger repercussions?

Reply:
discussed above